include/seqan3/range/container/aligned_sequence_adaptor_constant_access.hpp

```diff
@@ -0,0 +1,688 @@
+        gap_vector.resize(size);
+        for (size_type i = size_old; i < this->size(); ++i)
+            gap_vector[i] = 1;
+        // decrease compressed sequence eventually
+        if (size_old > 0)
+            sequence.resize(ss.select(size_old-1));
+        update_support_structures();
+    }
+    //{ val.resize(0, typename type::value_type{}) } -> void;
+    /*!\brief Resize the aligned sequence. Extend with gaps if the new sizes
+    * enlarges the current aligned sequence.
+    *
+    * Note that sdsl does not initialize int_vectors with 0s when applying resize.
+    */
+    void resize(size_type size, value_type value)
+    {
+        if (value == gap::GAP || size < this->size())
+            return resize(size);
+        // extend with non-gap value size()-size times
+        size_type size_old = this->size();
+        gap_vector.resize(size);
+        for (size_type i = size_old; i < this->size(); ++i)
+            gap_vector[i] = 0;
+        sequence.insert(sequence.cend(), this->size()-size_old, value);
+        update_support_structures();
+    }
+
+    //!\}
+private:
+    //!\privatesection
+    //!\brief Where the gap symbol is stored.
+    //!\hideinitializer
+    constexpr value_type static const gap_symbol = value_type(gap::GAP);
+    /*!\brief Gap presentation of the aligned sequence (1: gap, 0: alphabet letter).
+    *
+    * The bit vector size corresponds to the true gapped sequence size and is
+    * therefore used to answer queries about the gapped sequence size.
+    */
+    sdsl::bit_vector gap_vector;
+    /*!\brief Rank support structure for projection into ungapped sequence space.
+    *
+    * The rank of position i is number 1s in the prefix 0..i-1. This corresponds
+    * to the number of gaps in the aligned sequence. Rank queries are answered in
+    * constant time.
+    */
+    //!\hideinitializer
+    sdsl::rank_support_v5<1,1> rs = sdsl::rank_support_v5<1,1>(&gap_vector);
+    /*!\brief Select support structure for projection into gap space.
+    *
+    * Select i returns the gap vector position of the i-th zero in constant time.
+    */
+    //!\hideinitializer
+    sdsl::select_support_mcl<0,1> ss = sdsl::select_support_mcl<0,1>(&gap_vector);
+    /*!\brief Where the ungapped sequence is stored.
+    *
+    * Despite the 'compressed' representation of the sequence does not contain
+    * gap symbols, its value type is the union the actual alphabet type and the
+    * gap symbol. The reason behind is to avoid plenty of castings when assigning
+    * or querying for aligned sequence positions.
+    */
+    // TODO: maybe std::string for storage more efficient?
+    std::vector<value_type> sequence{};
+};
+
+// global swap { swap(val, val2) } -> void;, TODO: lhs and rhs same inner_type?
+template <typename inner_type, char gap_symbol = '_'>
+void swap (aligned_sequence_adaptor_constant_access<inner_type> & lhs, aligned_sequence_adaptor_constant_access<inner_type> & rhs)
```

Comment:
I don't think you need to add this explicity, std::swap is defined for everything that has move constructors IIRC.